src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -275,8 +277,9 @@ private BaseNamespaceDeclarationSyntax ParseNamespaceDeclarationCore(
             {
                 //the next character is neither the brace we expect, nor a token that could follow the expected
                 //brace so we assume it's a mistake and replace it with a missing brace 
-                openBrace = this.EatTokenWithPrejudice(SyntaxKind.OpenBraceToken);
-                openBrace = this.ConvertToMissingWithTrailingTrivia(openBrace, SyntaxKind.OpenBraceToken);
+                openBrace = this.ConvertToMissingWithTrailingTrivia(
+                    this.EatTokenWithPrejudiceForSkippedTokenList(SyntaxKind.OpenBraceToken),
+                    SyntaxKind.OpenBraceToken);
```

Comment:
EatTokenWithPrejudice has very special semantics (which i'venow documented). I've named it to make it much clearer as to when you should use it.